libvast/vast/value_index.hpp

```diff
@@ -98,9 +104,10 @@ class value_index {
   virtual caf::expected<ids>
   lookup_impl(relational_operator op, data_view x) const = 0;
 
-  ewah_bitmap mask_;      ///< The position of all values excluding nil.
-  ewah_bitmap none_;      ///< The positions of nil values.
-  const vast::type type_; ///< The type of this value index.
+  ewah_bitmap mask_;         ///< The position of all values excluding nil.
+  ewah_bitmap none_;         ///< The positions of nil values.
+  const vast::type type_;    ///< The type of this index.
+  const caf::settings opts_; ///< Runtime context with additional parameters.
```

Comment:
The following is intended as a start of discussion, not an outright request for change (depending on the outcome, we can still implement that in a follow up PR):

I'm not super happy about adding a direct relationship between `value_index` and `caf::settings` or any other polymorphic type. From a user point of view it does not make sense that they need to wrap the `length` parameter into a `caf::settings` so they can construct a `string_index`.

An alternative approach might be to leave `value_index` as it is, but to compose the `options` parameter to it by membership in a wrapper type:
```
// TODO: Find a better name ;).
struct value_index_wrapper {
  value_index_ptr index;
  caf::settings options;
};
```
I'm not sure if this really leads to a leaner implementation though.

Reply:
Yeah, I'm in agreement that this is not optimal. I don't like the tight coupling either, but I don't see another way to handle the polymorphic *recursive* type `value_index`. The problems are containers like `vector`, `set`, and `table`.